### ISSUE TITLE
Products (Beta) Block causes Critical Error when Google Listings & Ads is active and site uses PHP 8.0

### ISSUE THREAD

**Issue by masperber** [Automattic/wp-calypso#73115]
### Quick summary

On sites using PHP 8.0, the Products (Beta) Block causes a critical error when Google Listings & Ads is active. The products will not appear on the live page, and attempting to edit the page produces a critical error.

### Steps to reproduce

1. Use PHP 8.0 configuration on a test site.
2. Install and activate WooCommerce.
3. Add a Products (Beta) Block to a page and update the page.
4. Install and activate and Google Listnings & Ads

### What you expected to happen

The products should display on the page, and no critical error should be produced.

### What actually happened

The products are not displayed, and editing the page produces a critical error.

### Impact

All

### Available workarounds?

No and the platform is unusable

### Platform (Simple and/or Atomic)

Atomic

**Comment by dpasque** [comment]
I've been trying to reproduce this on a WoA site and a Jurassic Ninja site, and haven't had any luck so far unforuntately!

@masperber -- are you still seeing this on your ends? Are there maybe any extra details you know of that are needed to get this to happen?

**Comment by masperber** [comment]
Hi @dpasque ! :)

I've tried to reproduce on a test site, and I can no longer get this bug to reproduce. I also went back to the site that originally reported the bug and switched the site from PHP 7.4 to PHP 8.0, and the bug no longer manifests.

I have to conclude that some change to either WooCommerce or Google Listings & Ads that took place between 2023-02-08 and 2023-02-24 has resolved this bug.

I'll leave it up to you whether you wish to close this issue or investigate further.

Ref: p1675880449846489-slack-C03TY6J1A
